### PR TITLE
transfer blip prison here

### PR DESCRIPTION
**Pull request by bicatcho11**
## Description

transfer blip prison here

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

File: client/main.lua [hunk 1]
```diff
@@ -296,4 +296,23 @@ end)
...
+        SetBlipColour(blip, 42)
+        
+        BeginTextCommandSetBlipName('STRING')
+        AddTextComponentString(prisonStation.label)
+        EndTextCommandSetBlipName(blip)
+
+        if prisonStation.disable then
+            SetBlipDisplay(blip, 0)
+        end
+    end
```
Comment (solareon) [inline client/main.lua]: Place this in a function and add that function under the `QBCore:Client:OnPlayerLoaded` event handler.

File: client/main.lua [hunk 2]
```diff
@@ -296,4 +296,23 @@ end)
...
+        SetBlipScale(blip, 0.8)
+        SetBlipColour(blip, 42)
+        
+        BeginTextCommandSetBlipName('STRING')
+        AddTextComponentString(prisonStation.label)
+        EndTextCommandSetBlipName(blip)
+
+        if prisonStation.disable then
+            SetBlipDisplay(blip, 0)
+        end
```
Comment (solareon) [inline client/main.lua]: We don't need to disable the blip. If a user doesn't want a blip they can remove it from the configuration and the function should check if the table is nil before starting the loop

File: config.lua [hunk 3]
```diff
@@ -41,6 +41,9 @@ Config.Uniforms ={
 }
 
 Config.Locations = {
+    prison = {
+        {label = 'Prison', coords = vec3(1678.9, 2513.4, 45.6), disable = false},
```
Comment (solareon) [inline config.lua]: See comment on disabling the blips.

**Review by solareon — CHANGES_REQUESTED** [review]

File: client/main.lua [hunk 4]
```diff
@@ -296,4 +296,23 @@ end)
 ---@deprecated do not call.
 RegisterNetEvent('prison:client:canteen', function()
 	lib.print.error(GetInvokingResource(), "invoked deprecated prison:client:canteen event. No action taken.")
-end)
\ No newline at end of file
+end)
+
+CreateThread(function()
+    -- Create a marker for the prison station
```
Comment (solareon) [inline client/main.lua]:
```suggestion
    -- Create a marker for the prison station
    if not next(Config.Locations.prison) then return end
```

**Comment by bicatcho11** [comment]
@solareon Thank you for your cooperation to come up with a beautiful code

File: client/main.lua [hunk 5]
```diff
@@ -187,6 +187,17 @@ RegisterNetEvent('QBCore:Client:OnPlayerLoaded', function()
...
+		local blip = AddBlipForCoord(station.coords.x, station.coords.y, station.coords.z)
+		SetBlipSprite(blip, 188)
+		SetBlipAsShortRange(blip, true)
+		SetBlipScale(blip, 0.8)
+		SetBlipColour(blip, 42)
+		BeginTextCommandSetBlipName('STRING')
+		AddTextComponentString(station.label)
+		EndTextCommandSetBlipName(blip)
+	end
+
```
Comment (solareon) [inline client/main.lua]:
```suggestion
	createPrisonBlip()

```

Comment (solareon) [reply client/main.lua]: Then add this function up around line 59

```
local function createPrisonBlip()
	if not next(Config.Locations.prison) then return end
	for _, station in pairs(Config.Locations.prison) do
		local blip = AddBlipForCoord(station.coords.x, station.coords.y, station.coords.z)
		SetBlipSprite(blip, 188)
		SetBlipAsShortRange(blip, true)
		SetBlipScale(blip, 0.8)
		SetBlipColour(blip, 42)
		BeginTextCommandSetBlipName('STRING')
		AddTextComponentString(station.label)
		EndTextCommandSetBlipName(blip)
	end
end
```

**Review by solareon — CHANGES_REQUESTED** [review]

**Review by solareon — APPROVED** [review]

**Review by mafewtm — APPROVED** [review]

**Pull request merged**